src/vs/workbench/contrib/output/browser/outputServices.ts

```diff
@@ -196,7 +206,32 @@ export class OutputService extends Disposable implements IOutputService, ITextMo
 
 	private setActiveChannel(channel: OutputChannel | undefined): void {
 		this.activeChannel = channel;
-		this.activeFileOutputChannelContext.set(!!channel?.outputChannelDescriptor?.file);
+		const descriptor = channel?.outputChannelDescriptor;
+		this.activeFileOutputChannelContext.set(!!descriptor?.file);
+		this.activeOutputChannelLevelSettableContext.set(descriptor !== undefined && SetLogLevelAction.isLevelSettable(descriptor));
+		const setLevelContext = (): void => {
+			const channelLogLevel = descriptor?.log ? this.loggerService.getLogLevel(descriptor.file) : undefined;
+			this.activeOutputChannelLevelContext.set(channelLogLevel !== undefined ? LogLevelToString(channelLogLevel) : '');
+		};
+		const setLevelIsDefaultContext = async (): Promise<void> => {
+			const descriptor = this.activeChannel?.outputChannelDescriptor;
+			if (descriptor?.log) {
+				const channelLogLevel = this.loggerService.getLogLevel(descriptor.file);
+				const channelDefaultLogLevel = await this.defaultLogLevelsService.getDefaultLogLevel(descriptor.extensionId);
+				this.activeOutputChannelLevelIsDefaultContext.set(channelDefaultLogLevel === channelLogLevel);
+			} else {
+				this.activeOutputChannelLevelIsDefaultContext.set(false);
+			}
+		};
+		this._register(this.defaultLogLevelsService.onDidChangeDefaultLogLevels(() => {
+			setLevelIsDefaultContext();
+		}));
+		setLevelIsDefaultContext();
+		this._register(this.loggerService.onDidChangeLogLevel(_level => {
```

Comment:
Is not these listeners should be at class level? Otherwise, we will have these listeners everytime an active channel is set.